ports/servoshell/browser.rs

```diff
@@ -277,9 +277,13 @@ where
         self.event_queue.push(event);
     }
 
-    /// Returns true iff the caller needs to manually present a new frame.
-    pub fn handle_servo_events(&mut self, events: Vec<(Option<BrowserId>, EmbedderMsg)>) -> bool {
+    /// Returns true if the caller needs to manually present a new frame.
+    pub fn handle_servo_events(
+        &mut self,
+        events: Vec<(Option<BrowserId>, EmbedderMsg)>,
+    ) -> (bool, bool) {
```

Comment:
can we make this more self-documenting with a struct { need_present, history_changed }?